### PR TITLE
RING-NEST conversions

### PR DESCRIPTION
**Pull request by sbiquard**
I have implemented the pixel conversions between RING and NESTED orderings.

Below are some benchmarks on my machine (CPU-only) for 32 and 64-bit precision, and inputs of size 100_000 (nside=512 -> npix=3_145_728 for `reorder`).

chart-darkbackground-n100000

Caveats concerning `reorder`:
- it does not support masked arrays
- by default, its behavious is slightly different from healpy: it will reorder the full map in one go, instead of splitting it in chunks. This reduces computation time but uses more temporary memory. The healpy behaviour can be imitated by using the `process_by_chunks` argument.

### CODE REVIEW

File: jax_healpy/pixelfunc.py [hunk 1]
```diff
@@ -894,6 +902,307 @@ def _pix2ang_nest(nside: ArrayLike, ipix: ArrayLike) -> tuple[Array, Array]:
...
+    x = jnp.asarray(x)
+    y = jnp.asarray(y)
+    face = jnp.asarray(face)
+    if nest:
+        return _xyf2pix_nest(nside, x, y, face)
+    else:
+        return _xyf2pix_ring(nside, x, y, face)
+
+
+@partial(jit, static_argnames=['nside'])
```
[the remark this answers is not in the document]
Comment (sbiquard) [reply jax_healpy/pixelfunc.py]: Ok, c'est modifié

File: jax_healpy/pixelfunc.py [hunk 2]
```diff
@@ -894,6 +902,307 @@ def _pix2ang_nest(nside: ArrayLike, ipix: ArrayLike) -> tuple[Array, Array]:
...
+        val |= (ix & (1 << i)) << i
+        val |= (iy & (1 << i)) << (i + 1)
+        return val
+
+    # ix and iy are always less than nside, so there is no need to extract more bits than this
+    length = (nside - 1).bit_length()
+
+    # we use a native for loop because it seems faster than lax.fori_loop in this case
+    fpix = jnp.zeros_like(ix)
+    # fpix = lax.fori_loop(0, length, combine_bits, fpix, unroll=True)
```
[the remark this answers is not in the document]
Comment (sbiquard) [reply jax_healpy/pixelfunc.py]: Ok, modifié

File: jax_healpy/pixelfunc.py [hunk 3]
```diff
@@ -894,6 +902,307 @@ def _pix2ang_nest(nside: ArrayLike, ipix: ArrayLike) -> tuple[Array, Array]:
...
+
+    def combine_bits(i, val):
+        val |= (ix & (1 << i)) << i
+        val |= (iy & (1 << i)) << (i + 1)
+        return val
+
+    # ix and iy are always less than nside, so there is no need to extract more bits than this
+    length = (nside - 1).bit_length()
+
+    # we use a native for loop because it seems faster than lax.fori_loop in this case
```
[the remark this answers is not in the document]
Comment (sbiquard) [reply jax_healpy/pixelfunc.py]: Ça me convient !
